src/dynlib/dynlib_windows.c

```diff
@@ -37,6 +37,46 @@ static char *generate_file_name(const char *name, uint32_t major_ver, uint32_t m
     return versioned_file_name;
 }
 
+static DLL_DIRECTORY_COOKIE add_current_module_to_search()
+{
+    wchar_t path[MAX_PATH];
+    HMODULE hModule = NULL;
+
+    if (GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS, (LPCTSTR)add_current_module_to_search, &hModule) == 0)
+    {
+        LOG_WARNING("Failed to get current module (%d).", GetLastError());
+        return NULL;
+    }
+
+    if (GetModuleFileNameW(hModule, path, sizeof(path)) == 0)
+    {
+        LOG_WARNING("Failed to get current module file name (%d).", GetLastError());
+        return NULL;
+    }
+
+    // GetModuleFileName give the full path, but AddDllDirectory requires a directory path. Remove
```

Comment:
What if we used [PathCchRemoveFileSpec](https://docs.microsoft.com/en-us/windows/win32/api/pathcch/nf-pathcch-pathcchremovefilespec) instead of manually stripping off k4a.dll? That way we don't need to worry what the module name is.

Reply:
Sounds good.